eng/pipelines/libraries/build-job.yml

```diff
@@ -0,0 +1,122 @@
+parameters:
+  buildConfig: ''
+  osGroup: ''
+  osSubgroup: ''
+  archType: ''
+  framework: netcoreapp
+  isOfficialBuild: false
+  timeoutInMinutes: 150
+  preBuildSteps: []
+  container: ''
+  variables: {}
+  pool: ''
+  runTests: false
+  testScope: ''
+
+jobs:
+  - template: /eng/pipelines/libraries/base-job.yml
+    parameters:
+      buildConfig: ${{ parameters.buildConfig }}
+      osGroup:  ${{ parameters.osGroup }}
+      osSubgroup:  ${{ parameters.osSubgroup }}
+      archType:  ${{ parameters.archType }}
+      framework:  ${{ parameters.framework }}
+      isOfficialBuild: ${{ parameters.isOfficialBuild }}
+      timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+      preBuildSteps: ${{ parameters.preBuildSteps }}
+      container: ${{ parameters.container }}
+      pool: ${{ parameters.pool }}
+      testScope: ${{ parameters.testScope }}
+      stepName: Build
+
+      variables:
+        - _skipTestRestoreArg: /p:SkipTestRestore=true
+        - _buildAction: -build
+        - ${{ parameters.variables }}
+        - ${{ if eq(parameters.osGroup, 'WebAssembly') }}:
+          - EMSDK_PATH: $(Build.BinariesDirectory)/emsdk
+        - ${{ if eq(parameters.runTests, true) }}:
+          - _buildAction: -build -buildtests /p:ArchiveTests=true
+          - ${{ if ne(parameters.framework, 'allConfigurations') }}:
+            - _skipTestRestoreArg: /p:SkipTestRestore=false
+
+      steps:
+        - script: $(_buildScript) -restore $(_buildArguments) $(_skipTestRestoreArg)
+          displayName: Restore
+
+        - ${{ if eq(parameters.isOfficialBuild, true) }}:
+          - task: DotNetCoreCLI@2
+            displayName: Restore internal tools
```

Comment:
I think it's worth discussing if we should extract that into the common layer in a subsequent PR.

Reply:
Related: https://github.com/dotnet/runtime/blob/master/eng/install-nuget-credprovider.sh

Reply:
Yeah I think this should definitely go into the common layer whenever we do the official build.